api/agent/drivers/docker/docker.go

```diff
@@ -102,6 +130,22 @@ func NewDocker(conf drivers.Config) *DockerDriver {
 		}
 	}
 
+	imageEvictChan := make(chan docker.APIImages, 10)
+	onEvictor := OnImageEvict(imageEvictChan)
+	go NewImageCleaner(driver, imageEvictChan)
+
+	logrus.Infof("Max cache size: %v\n", conf.MaxImageCacheSize)
+	driver.imageLRU = NewLRU(int64(conf.MaxImageCacheSize), onEvictor)
+
+	go func() {
```

Comment:
so, if the go-routine completes after loadDockerImages() below, we might accidentally clear on-disk requested images.